### PR TITLE
Dotnet steps

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* `withDotnet` step
* `withTerraform` step

## Why is it important?

these new steps make easier to install and use dotnet SDK and Terraform.

### CODE REVIEW

File: vars/withTerraform.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+ Install .NET SDK and export environment variables to use it.
+
+withDotnet(root: "${env.WORKSPACE}/bin"){
```
Comment (mdelapenya) [inline vars/withTerraform.groovy:21]:
```suggestion
Install Terraform from a GCS bucket and export environment variables to use it.

withTerraform(root: "${env.WORKSPACE}/bin", version: "0.15.3"){
```

**Review by mdelapenya — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by v1v — COMMENTED** [review]
Do you think we could merge this implementation with https://github.com/elastic/apm-pipeline-library/blob/master/vars/installTools.groovy ?

**Comment by kuisathaverat** [comment]
> Do you think we could merge this implementation with https://github.com/elastic/apm-pipeline-library/blob/master/vars/installTools.groovy ?

yep, the scripts can be reused to make an install step on `installTools`, in a follow up we can make the refactor.

**Comment by v1v** [comment]
write something

**Pull request closed without merging**